winbuild/build_prepare.py

```diff
@@ -55,24 +56,28 @@ def cmd_nmake(makefile=None, target="", params=None):
     )
 
 
-def cmd_cmake(params=None, file="."):
-    if params is None:
-        params = ""
-    elif isinstance(params, (list, tuple)):
-        params = " ".join(params)
-    else:
-        params = str(params)
-    return " ".join(
-        [
-            "{cmake}",
-            "-DCMAKE_VERBOSE_MAKEFILE=ON",
-            "-DCMAKE_RULE_MESSAGES:BOOL=OFF",
-            "-DCMAKE_BUILD_TYPE=Release",
-            f"{params}",
-            '-G "NMake Makefiles"',
-            f'"{file}"',
-        ]
-    )
+def cmds_cmake(target, *params):
+    if not isinstance(target, str):
+        target = " ".join(target)
+
+    return [
+        " ".join(
+            [
+                "{cmake}",
+                "-DCMAKE_BUILD_TYPE=Release",
+                "-DCMAKE_VERBOSE_MAKEFILE=ON",
+                "-DCMAKE_RULE_MESSAGES:BOOL=OFF",  # for NMake
+                "-DCMAKE_C_COMPILER=cl.exe",  # for Ninja
+                "-DCMAKE_CXX_COMPILER=cl.exe",  # for Ninja
+                "-DCMAKE_C_FLAGS=-nologo",
+                "-DCMAKE_CXX_FLAGS=-nologo",
+                *params,
+                '-G "{cmake_generator}"',
+                ".",
+            ]
+        ),
+        f"{{cmake}} --build . --clean-first --parallel --target {target}",
```

Comment:
I originally updated the documentation in this PR to require CMake 3.13+ as that is what xz's CMake files specify.

I missed that the `cmake --build` command requires CMake 3.15+ when there are multiple targets, I've now updated the PR.

The previous CMake requirement was 3.12, which is what is included with VS2017 and was the oldest I tested when writing the documentation in #4495.

VS2019 includes CMake 3.20, but I've found that 3.15 is sufficient for this PR.